internal/character/clara/trace.go

```diff
@@ -0,0 +1,64 @@
+package clara
+
+import (
+	"github.com/simimpact/srsim/pkg/engine/event"
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/modifier"
+	"github.com/simimpact/srsim/pkg/engine/prop"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+// A2:
+// When attacked, this character has a 35.00 % fixed chance to remove a debuff placed on them.
+// A4:
+// The chance to resist Crowd Control Debuffs increases by 35.00 %.
+// A6:
+// Increases Svarog's Counter DMG by 30.00 %.
+
+const (
+	A2 key.Modifier = "clara-a2"
+	A4 key.Modifier = "clara-a4"
+	A6 key.Modifier = "clara-a6"
+)
+
+func init() {
+	modifier.Register(A2, modifier.Config{
+		Listeners: modifier.Listeners{
+			OnBeforeBeingAttacked: func(mod *modifier.Instance, e event.AttackStart) {
+				if mod.Engine().Rand().Float32() < 0.35 {
+					mod.Engine().DispelStatus(mod.Owner(), info.Dispel{
+						Status: model.StatusType_STATUS_DEBUFF,
+						Order:  model.DispelOrder_LAST_ADDED,
+						Count:  1,
+					})
+				}
+			},
+		},
+	})
+}
+
+func (c *char) initTraces() {
+	if c.info.Traces["1107101"] {
+		c.engine.AddModifier(c.id, info.Modifier{
+			Name:   A2,
+			Source: c.id,
+		})
+	}
+
+	if c.info.Traces["1107102"] {
+		c.engine.AddModifier(c.id, info.Modifier{
+			Name:      A4,
+			Source:    c.id,
+			DebuffRES: info.DebuffRESMap{model.BehaviorFlag_STAT_CTRL: 0.35},
+		})
+	}
+
+	if c.info.Traces["1107103"] {
+		c.engine.AddModifier(c.id, info.Modifier{
+			Name:   A6,
+			Source: c.id,
+			Stats:  info.PropMap{prop.AllDamagePercent: 0.3},
+		})
+	}
```

Comment:
This is incorrect. In current implementation, this will buff **all** Clara attacks by 30%

datamine implementation does the following:
1. modifier implemented with `OnBeforeHit` listener. Listener modifies hit data to give attacker 30% `AllDamagePercent` increase
2. modifier is added to clara at the start of counter
3. modifier is removed from clara at the end of counter

I think we can simplify the implementation to the following:
1. modifier implements `OnBeforeHit` listener
2. In `OnBeforeHit` listener, modify hit `AllDamagePercent` iff `AttackType == INSERT`
3. Apply this modifier on init (always on like the other two)

This variant should work and be equivalent since the only attacks by Clara that are `AttackType == INSERT` are her counter attacks